### PR TITLE
Add %json tags #44

### PR DESCRIPTION
**Pull request by gabeta**
Added compiler to encode json.

### CODE REVIEW

**Comment by papac** [comment]
@gabeta can you please run `composer phpcbf` to normalize the coding standard?

File: tests/CompileJsonTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    private Compiler $compiler;
+
+    public function setUp(): void
+    {
+        $this->compiler = new Compiler();
+    }
+
+    public function testCompileJson()
+    {
+        $compile_raw_php = $this->makeReflectionFor('compileJson');
```
Comment (papac) [inline tests/CompileJsonTest.php]: Rename the variable name

File: tests/CompileJsonTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        $render = $compile_raw_php->invoke($this->compiler, "%json(['name' => 'tintin'])");
+
+        $this->assertEquals($render, "<?php echo json_encode(['name' => 'tintin']); ?>");
+    }
+
+    public function testCompileJsonWithEncodingOptions()
+    {
+        $compile_raw_php = $this->makeReflectionFor('compileJson');
```
Comment (papac) [inline tests/CompileJsonTest.php]: Rename the variable `$compile_raw_php` name to understanding

File: src/Lexique/CompileJson.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Compile the %json statement
+     *
+     * @param string $expression
+     * @return string
+     */
+    protected function compileJson(string $expression): string
+    {
+        $output = preg_replace_callback(
+            '/^%json\((.*)\)$/',
```
Comment (papac) [inline src/Lexique/CompileJson.php]: Sometimes the developers write directives like `%json ()`. You must consider the white spacing between `%json ` and `()`

**Review by papac — CHANGES_REQUESTED** [review]

File: src/Lexique/CompileJson.php (hunk 3 above)
Comment (gabeta) [reply src/Lexique/CompileJson.php]: Thanks, I forgot that detail.

**Review by papac — APPROVED** [review]

**Pull request merged**